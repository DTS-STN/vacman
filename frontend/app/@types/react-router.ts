import 'react-router';

import type { Namespace } from 'i18next';

declare module 'react-router' {
  interface AppLoadContext {
    nonce: string;
    session: AppSession;
  }

  /**
   * Route handles should export an i18n namespace, if necessary.
   */
  interface RouteHandle {
    i18nNamespace?: Namespace;
  }

  /**
   * A route module exports an optional RouteHandle.
   */
  interface RouteModule {
    handle?: RouteHandle;
  }

  /**
   * Override the default React Router RouteModules
   * to include the new RouteModule type.
   */
  interface RouteModules extends Record<string, RouteModule | undefined> {}
}

export {};
