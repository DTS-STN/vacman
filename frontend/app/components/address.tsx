import type { ComponentProps, JSX } from 'react';

import { formatAddress } from '~/utils/string-utils';
import { cn } from '~/utils/tailwind-utils';

type AddressProps = ComponentProps<'address'> & {
  address: Parameters<typeof formatAddress>[0];
  format?: Parameters<typeof formatAddress>[1];
};

export function Address({ address, format, className, ...rest }: AddressProps): JSX.Element {
  return (
    <address className={cn('whitespace-pre-wrap not-italic', className)} data-testid="address-id" {...rest}>
      {formatAddress(address, format)}
    </address>
  );
}
