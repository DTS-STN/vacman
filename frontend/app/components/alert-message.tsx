import type { Ref } from 'react';

import type { IconDefinition } from '@fortawesome/free-solid-svg-icons';
import { faTriangleExclamation, faCircleExclamation, faCircleCheck } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

interface AlertMessageProps {
  message: string;
  type: 'success' | 'info' | 'error';
  ref?: Ref<HTMLDivElement>;
}

const styles: Record<NonNullable<AlertMessageProps['type']>, string> = {
  success: `border-green-600 bg-green-100`,
  error: `border-red-800 bg-red-100`,
  info: `border-sky-700 bg-sky-100`,
};

const icons: Record<NonNullable<AlertMessageProps['type']>, IconDefinition> = {
  success: faCircleCheck,
  error: faTriangleExclamation,
  info: faCircleExclamation,
};

export function AlertMessage({ ref, message, type = 'info' }: AlertMessageProps) {
  return (
    <div ref={ref} className={`${styles[type]} flex w-full items-center border-l-4 p-2`}>
      <FontAwesomeIcon
        icon={icons[type]}
        className={`${type === 'success' ? 'text-green-600' : type === 'error' ? 'text-red-800' : 'text-sky-700'} mr-2`}
      />
      {message}
    </div>
  );
}
