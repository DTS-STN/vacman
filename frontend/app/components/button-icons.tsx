import type { FontAwesomeIconProps } from '@fortawesome/react-fontawesome';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { cn } from '~/utils/tailwind-utils';

export function ButtonStartIcon({ className, ...restProps }: FontAwesomeIconProps) {
  return <FontAwesomeIcon fixedWidth className={cn('me-1', className)} {...restProps} />;
}

export function ButtonEndIcon({ className, ...restProps }: FontAwesomeIconProps) {
  return <FontAwesomeIcon fixedWidth className={cn('ms-1', className)} {...restProps} />;
}
