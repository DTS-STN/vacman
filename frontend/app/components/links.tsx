import type { ComponentProps, JSX, MouseEvent } from 'react';

import type { Params, Path } from 'react-router';
import { generatePath, Link } from 'react-router';

import { useTranslation } from 'react-i18next';

import { AppError } from '~/errors/app-error';
import { ErrorCodes } from '~/errors/error-codes';
import { useLanguage } from '~/hooks/use-language';
import type { I18nRouteFile } from '~/i18n-routes';
import { i18nRoutes } from '~/i18n-routes';
import { getRouteByFile } from '~/utils/route-utils';
import { cn } from '~/utils/tailwind-utils';

/**
 * Props for a bilingual link component targeting an internationalized route.
 *
 * This type extends the base properties of a React Router `Link` component,
 * excluding the `to` property, and adds fields specific to i18n routes.
 */
type BilingualLinkProps = OmitStrict<ComponentProps<typeof Link>, 'to'> & {
  disabled?: boolean;
  file: I18nRouteFile;
  hash?: Path['hash'];
  lang?: Language;
  params?: Params;
  search?: Path['search'];
  to?: never;
  newTabIndicator?: boolean;
};

/**
 * Props for a unilingual link component targeting a route in a single language.
 *
 * This type extends the base properties of a React Router `Link` component,
 * excluding the `lang` property from the base type, and enforces constraints
 * to ensure that only standard routes (non-i18n) are used.
 */
type UnilingualLinkProps = OmitStrict<ComponentProps<typeof Link>, 'lang'> & {
  disabled?: boolean;
  file?: never;
  hash?: never;
  lang?: Language;
  params?: never;
  search?: never;
  newTabIndicator?: boolean;
};

/**
 * Props for the `AppLink` component, supporting both bilingual (i18n) and unilingual routes.
 */
type AppLinkProps = BilingualLinkProps | UnilingualLinkProps;

/**
 * AnchorLinkProps represents the properties for the AnchorLink component.
 * It extends the ComponentProps<'a'> type, omitting the 'href' property,
 * and adds the required 'anchorElementId' property.
 */
type AnchorLinkProps = OmitStrict<ComponentProps<'a'>, 'href'> & {
  anchorElementId: string;
};

/**
 * Props for the `InlineLink` component. Effecetively extends the `AppLink` component's props.
 */
type InlineLinkProps = ComponentProps<typeof AppLink>;

/**
 * A flexible link component that supports both standard and i18n routes.
 *
 * - If the `to` property is provided, it creates a standard react-router link to the specified target.
 * - If the `to` property is omitted, the link is constructed dynamically using the provided `file`, `lang`, `params`, and other properties.
 *
 * @throws If the `lang` parameter is not provided and the current language cannot be determined.
 */
export function AppLink({
  children,
  disabled,
  hash,
  lang,
  params,
  file,
  search,
  to,
  newTabIndicator,
  ...props
}: AppLinkProps): JSX.Element {
  const { currentLanguage } = useLanguage();

  const newTabProps = newTabIndicator ? { target: '_blank', rel: 'noopener noreferrer' } : {};

  if (to !== undefined) {
    return (
      <Link lang={lang} to={to} {...newTabProps} {...props}>
        {children}
        {newTabIndicator && <NewTabIndicator />}
      </Link>
    );
  }

  const targetLanguage = lang ?? currentLanguage;

  if (targetLanguage === undefined) {
    throw new AppError(
      'The `lang` parameter was not provided, and the current language could not be determined from the request',
      ErrorCodes.MISSING_LANG_PARAM,
    );
  }

  if (disabled) {
    return (
      // see: "disabling a link" -- https://www.scottohara.me/blog/2021/05/28/disabled-links.html
      <a role="link" aria-disabled="true" {...newTabProps} {...props}>
        {children}
        {newTabIndicator && <NewTabIndicator />}
      </a>
    );
  }

  const route = getRouteByFile(file, i18nRoutes);
  const pathname = generatePath(route.paths[targetLanguage], params);

  const langProp = targetLanguage !== currentLanguage ? targetLanguage : undefined;
  const reloadDocumentProp = props.reloadDocument ?? lang !== undefined;

  return (
    <Link lang={langProp} to={{ hash, pathname, search }} reloadDocument={reloadDocumentProp} {...props}>
      {children}
      {newTabIndicator && <NewTabIndicator />}
    </Link>
  );
}

/**
 * AnchorLink is a React component used to create anchor links that scroll
 * and focus on the specified target element when clicked.
 *
 * @param props - The properties for the AnchorLink component.
 * @returns React element representing the anchor link.
 */
export function AnchorLink({ anchorElementId, children, onClick, ...restProps }: AnchorLinkProps): JSX.Element {
  /**
   * handleOnSkipLinkClick is the click event handler for the anchor link.
   * It prevents the default anchor link behavior, scrolls to and focuses
   * on the target element specified by 'anchorElementId', and invokes
   * the optional 'onClick' callback.
   */
  function handleOnSkipLinkClick(event: MouseEvent<HTMLAnchorElement>) {
    event.preventDefault();
    scrollAndFocusFromAnchorLink(event.currentTarget.href);
    onClick?.(event);
  }

  return (
    <a href={`#${anchorElementId}`} onClick={handleOnSkipLinkClick} data-testid="anchor-link" {...restProps}>
      {children}
    </a>
  );
}

/**
 * A styled link component designed for inline text links, supporting both standard and i18n routes.
 *
 * - For i18n routes, the `file`, `hash`, `params`, and `search` properties are used to dynamically construct the URL.
 * - For standard routes, the `to` property is used directly.
 * - Applies default styling with additional customization via the `className` property.
 *
 * @throws If the `lang` parameter is not provided and the current language cannot be determined.
 */
export function InlineLink({
  className,
  children,
  file,
  hash,
  params,
  search,
  to,
  newTabIndicator,
  ...props
}: InlineLinkProps): JSX.Element {
  const baseClassName = cn('text-slate-700 underline hover:text-blue-700 focus:text-blue-700');

  if (file) {
    return (
      <AppLink
        className={cn(baseClassName, className)}
        file={file}
        hash={hash}
        params={params}
        search={search}
        newTabIndicator={newTabIndicator}
        {...props}
      >
        {children}
      </AppLink>
    );
  }

  return (
    <AppLink className={cn(baseClassName, className)} to={to} newTabIndicator={newTabIndicator} {...props}>
      {children}
    </AppLink>
  );
}

/**
 * Scrolls and focuses on the element identified by the anchor link's hash.
 *
 * @param href - The anchor link URL.
 */
function scrollAndFocusFromAnchorLink(href: string): void {
  if (URL.canParse(href)) {
    const { hash } = new URL(href);

    if (hash) {
      const targetElement = document.getElementById(hash.replace('#', ''));

      if (targetElement) {
        targetElement.scrollIntoView({ behavior: 'smooth' });
        targetElement.focus();
      }
    }
  }
}

function NewTabIndicator({ className, ...props }: OmitStrict<ComponentProps<'span'>, 'children'>): JSX.Element {
  const { t } = useTranslation('gcweb');
  // Following whitespace is important to ensure the content's text is seperated for the screen-reader text
  return (
    <span className={cn('sr-only', className)} {...props}>
      {` (${t('screen-reader.new-tab')})`}
    </span>
  );
}
