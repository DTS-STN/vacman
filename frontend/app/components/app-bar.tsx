import type { JSX } from 'react';

import { faChevronDown, faRightFromBracket, faUser } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { useTranslation } from 'react-i18next';

import { DropdownMenu, DropdownMenuContent, DropdownMenuLabel, DropdownMenuTrigger } from '~/components/dropdown-menu';
import { LanguageSwitcher } from '~/components/language-switcher';
import { Menu, MenuItem } from '~/components/menu';
import { cn } from '~/utils/tailwind-utils';

type AppBarProps = {
  children: React.ReactNode;
  name?: string;
  profileItems?: React.ReactNode;
};

export function AppBar({ children, name, profileItems }: AppBarProps): JSX.Element {
  const { t } = useTranslation(['gcweb']);

  return (
    <div className="bg-slate-700">
      <div className="align-center container mx-auto flex flex-wrap justify-between">
        <div className="align-center flex">
          <Menu>{children}</Menu>
        </div>
        <div className="flex items-center space-x-4 text-right">
          {name && <UserButton name={name}>{profileItems}</UserButton>}
          <LanguageSwitcher>{t('gcweb:language-switcher.alt-lang')}</LanguageSwitcher>
        </div>
      </div>
    </div>
  );
}

type UserButtonProps = {
  className?: string;
  children?: React.ReactNode;
  name?: string;
};

function UserButton({ className, children, name }: UserButtonProps): JSX.Element {
  const { t } = useTranslation(['gcweb']);

  return (
    <DropdownMenu>
      <DropdownMenuTrigger
        className={cn(
          'flex h-full flex-nowrap space-x-2 bg-slate-600 px-2 text-sm text-white hover:bg-slate-500 hover:underline focus:ring-2 focus:ring-black focus:ring-offset-2 focus:outline-hidden aria-expanded:bg-slate-800 aria-expanded:text-white sm:space-x-4 sm:px-4',
          className,
        )}
      >
        <div className="text-md my-auto flex flex-nowrap items-center space-x-2 py-2">
          <div className="flex h-8 w-8 items-center justify-center rounded-full bg-white">
            <FontAwesomeIcon icon={faUser} className="size-5 text-slate-700" />
          </div>
          <span id="menu-label" className="text-md hidden py-2 font-bold sm:block">
            {name}
          </span>
        </div>
        <FontAwesomeIcon icon={faChevronDown} className="my-auto size-4" />
      </DropdownMenuTrigger>
      <DropdownMenuContent align="end" className="w-64 bg-slate-700">
        <UserName name={name} />
        <MenuItem file="routes/protected/index.tsx">{t('gcweb:app.profile')}</MenuItem>
        {children}
        <MenuItem to="/auth/logout" className="text-md flex justify-between">
          {t('gcweb:app.logout')}
          <FontAwesomeIcon icon={faRightFromBracket} className="my-auto size-8" />
        </MenuItem>
      </DropdownMenuContent>
    </DropdownMenu>
  );
}

type UserNameProps = {
  name?: string;
};

function UserName({ name }: UserNameProps): JSX.Element {
  return (
    <>
      {name !== undefined && (
        <DropdownMenuLabel className="text-md flex items-center border-b-2 border-slate-600 px-3 py-2 text-gray-300 sm:hidden">
          <FontAwesomeIcon icon={faUser} className="mr-2 size-4" />
          {name}
        </DropdownMenuLabel>
      )}
    </>
  );
}
