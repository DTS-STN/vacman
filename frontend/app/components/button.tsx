import type { ComponentProps } from 'react';

import { ButtonEndIcon, ButtonStartIcon } from '~/components/button-icons';
import { cn } from '~/utils/tailwind-utils';

type ButtonEndIconProps = ComponentProps<typeof ButtonEndIcon>;
type ButtonStartIconProps = ComponentProps<typeof ButtonStartIcon>;

const sizes = {
  xs: 'px-3 py-2 text-xs',
  sm: 'px-3 py-2 text-sm',
  base: 'px-5 py-2.5 text-sm',
  lg: 'px-5 py-3 text-base',
  xl: 'px-6 py-3.5 text-base',
} as const;

// prettier-ignore
const variants = {
  alternative: 'border-gray-200 bg-white text-gray-900 hover:bg-gray-100 hover:text-blue-700 focus:bg-gray-100 focus:text-blue-700',
  default: 'border-gray-300 bg-gray-200 text-slate-700 hover:bg-neutral-300 focus:bg-neutral-300',
  dark: 'border-gray-800 bg-gray-800 text-white hover:bg-gray-900 focus:bg-gray-900',
  green: 'border-green-700 bg-green-700 text-white hover:bg-green-800 focus:bg-green-800',
  primary: 'border-slate-700 bg-slate-700 text-white hover:bg-sky-800 focus:bg-sky-800',
  red: 'border-red-700 bg-red-700 text-white hover:bg-red-800 focus:bg-red-800',
  link: 'border-0 bg-white text-gray-900 underline hover:bg-gray-100 hover:text-blue-700 focus:bg-gray-100 focus:text-blue-700',
  ghost: 'border-0 hover:bg-neutral-100 hover:text-neutral-900',
  outline: "border border-neutral-200 bg-white hover:bg-neutral-100 hover:text-neutral-900 disabled:text-neutral-500",
} as const;

export interface ButtonProps extends ComponentProps<'button'> {
  endIcon?: ButtonEndIconProps['icon'];
  endIconProps?: OmitStrict<ButtonEndIconProps, 'icon'>;
  pill?: boolean;
  size?: keyof typeof sizes;
  startIcon?: ButtonStartIconProps['icon'];
  startIconProps?: OmitStrict<ButtonStartIconProps, 'icon'>;
  variant?: keyof typeof variants;
}

/**
 * Tailwind CSS Buttons from Flowbite
 * @see https://flowbite.com/docs/components/buttons/
 */
export function Button({
  children,
  className,
  endIcon,
  endIconProps,
  pill,
  size = 'base',
  startIcon,
  startIconProps,
  variant = 'default',
  ...props
}: ButtonProps) {
  const buttonClassName = cn(
    'font-lato inline-flex items-center justify-center rounded border align-middle outline-offset-4',
    sizes[size],
    variants[variant],
    'disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70',
    pill && 'rounded-full',
    className,
  );
  return (
    <button className={buttonClassName} {...props}>
      {startIcon && <ButtonStartIcon {...(startIconProps ?? {})} icon={startIcon} />}
      {children}
      {endIcon && <ButtonEndIcon {...(endIconProps ?? {})} icon={endIcon} />}
    </button>
  );
}
