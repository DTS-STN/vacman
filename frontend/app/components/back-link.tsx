import type { Params, Path } from 'react-router';

import { faAngleLeft } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { useTranslation } from 'react-i18next';

import { InlineLink } from '~/components/links';
import type { I18nRouteFile } from '~/i18n-routes';
import { cn } from '~/utils/tailwind-utils';

interface BackLinkProps {
  file: I18nRouteFile;
  params?: Params;
  className?: string;
  hash?: Path['hash'];
  search?: Path['search'];
  translationKey?: string;
}

export function BackLink({ className, file, params, hash, search, translationKey = 'app:profile.back' }: BackLinkProps) {
  const { t } = useTranslation();
  const linkText = t(translationKey);
  return (
    <InlineLink
      className={cn('inline-flex items-center', className)}
      file={file}
      params={params}
      hash={hash}
      search={search}
      aria-label={linkText}
    >
      <FontAwesomeIcon icon={faAngleLeft} />
      {linkText}
    </InlineLink>
  );
}
