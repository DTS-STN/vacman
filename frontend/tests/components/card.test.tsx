import { createRoutesStub } from 'react-router';

import { faUser } from '@fortawesome/free-solid-svg-icons';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { Button } from '~/components/button';
import {
  Card,
  CardContent,
  CardDescription,
  CardFooter,
  CardHeader,
  CardIcon,
  CardImage,
  CardTag,
  CardTitle,
} from '~/components/card';
import { AppLink } from '~/components/links';

describe('Card', () => {
  it('should render a card', () => {
    const { container } = render(
      <Card>
        <CardHeader>
          <CardTitle>Card Title</CardTitle>
          <CardDescription>Test Card</CardDescription>
        </CardHeader>
      </Card>,
    );
    expect(container).toMatchSnapshot('expected html');
  });

  it('should render a card with a link', () => {
    const RoutesStub = createRoutesStub([
      {
        path: '/fr/public',
        Component: () => (
          <Card asChild>
            <AppLink file="routes/public/index.tsx">
              <CardHeader>
                <CardTitle>Card Title</CardTitle>
                <CardDescription>Test Card</CardDescription>
              </CardHeader>
            </AppLink>
          </Card>
        ),
      },
    ]);

    const { container } = render(<RoutesStub initialEntries={['/fr/public']} />);
    expect(container).toMatchSnapshot('expected html');
  });

  it('should render a card with a link with an image', () => {
    const RoutesStub = createRoutesStub([
      {
        path: '/fr/public',
        Component: () => (
          <Card asChild>
            <AppLink file="routes/public/index.tsx">
              <CardImage src="https://www.canada.ca/content/dam/canada/activities/20250115-1-520x200.jpg" />
              <CardHeader>
                <CardTitle>Card Title</CardTitle>
                <CardDescription>Test Card</CardDescription>
              </CardHeader>
              <CardFooter>
                <CardTag>Coming soon</CardTag>
              </CardFooter>
            </AppLink>
          </Card>
        ),
      },
    ]);

    const { container } = render(<RoutesStub initialEntries={['/fr/public']} />);
    expect(container).toMatchSnapshot('expected html');
  });

  it('should render a card with a link with an icon', () => {
    const RoutesStub = createRoutesStub([
      {
        path: '/fr/public',
        Component: () => (
          <Card>
            <div className="flex items-center gap-4 p-6">
              <CardIcon icon={faUser} />
              <CardHeader className="p-0">
                <CardTitle asChild>
                  <h2>Card Title</h2>
                </CardTitle>
                <CardDescription>Test Card</CardDescription>
              </CardHeader>
            </div>
          </Card>
        ),
      },
    ]);

    const { container } = render(<RoutesStub initialEntries={['/fr/public']} />);
    expect(container).toMatchSnapshot('expected html');
  });

  it('should render a card with a link with an icon, content, footer with button and heading2', () => {
    const RoutesStub = createRoutesStub([
      {
        path: '/fr/public',
        Component: () => (
          <Card>
            <div className="flex items-center gap-4 p-6">
              <CardIcon icon={faUser} />
              <CardHeader className="p-0">
                <CardTitle asChild>
                  <h2>Card Title</h2>
                </CardTitle>
                <CardDescription>Test Card</CardDescription>
              </CardHeader>
            </div>
            <CardContent className="space-y-3">
              <p>Card content part 1</p>
              <p>Card content part 2</p>
            </CardContent>
            <CardFooter>
              <Button>Click me</Button>
            </CardFooter>
          </Card>
        ),
      },
    ]);

    const { container } = render(<RoutesStub initialEntries={['/fr/public']} />);
    expect(container).toMatchSnapshot('expected html');
  });
});
