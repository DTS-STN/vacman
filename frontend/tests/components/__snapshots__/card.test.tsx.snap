// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > should render a card > expected html 1`] = `
<div>
  <div
    class="group block rounded-xs border bg-white shadow-xs [&:is(a)]:hover:bg-gray-50 [&:is(a)]:hover:shadow-md"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
    >
      <h3
        class="leading-none font-semibold group-[&:is(a)]:text-slate-700 group-[&:is(a)]:underline group-[&:is(a):focus]:text-blue-700 group-[&:is(a):hover]:text-blue-700"
      >
        Card Title
      </h3>
      <div
        class="text-sm text-black/60"
      >
        Test Card
      </div>
    </div>
  </div>
</div>
`;

exports[`Card > should render a card with a link > expected html 1`] = `
<div>
  <a
    class="group block rounded-xs border bg-white shadow-xs [&:is(a)]:hover:bg-gray-50 [&:is(a)]:hover:shadow-md"
    data-discover="true"
    href="/fr"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
    >
      <h3
        class="leading-none font-semibold group-[&:is(a)]:text-slate-700 group-[&:is(a)]:underline group-[&:is(a):focus]:text-blue-700 group-[&:is(a):hover]:text-blue-700"
      >
        Card Title
      </h3>
      <div
        class="text-sm text-black/60"
      >
        Test Card
      </div>
    </div>
  </a>
</div>
`;

exports[`Card > should render a card with a link with an icon > expected html 1`] = `
<div>
  <div
    class="group block rounded-xs border bg-white shadow-xs [&:is(a)]:hover:bg-gray-50 [&:is(a)]:hover:shadow-md"
  >
    <div
      class="flex items-center gap-4 p-6"
    >
      <div
        class="flex size-10 shrink-0 items-center justify-center rounded-full bg-slate-700 sm:size-12"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-user size-4 text-white sm:size-5"
          data-icon="user"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512l388.6 0c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304l-91.4 0z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="flex flex-col space-y-1.5 p-0"
      >
        <h2
          class="leading-none font-semibold group-[&:is(a)]:text-slate-700 group-[&:is(a)]:underline group-[&:is(a):focus]:text-blue-700 group-[&:is(a):hover]:text-blue-700"
        >
          Card Title
        </h2>
        <div
          class="text-sm text-black/60"
        >
          Test Card
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Card > should render a card with a link with an icon, content, footer with button and heading2 > expected html 1`] = `
<div>
  <div
    class="group block rounded-xs border bg-white shadow-xs [&:is(a)]:hover:bg-gray-50 [&:is(a)]:hover:shadow-md"
  >
    <div
      class="flex items-center gap-4 p-6"
    >
      <div
        class="flex size-10 shrink-0 items-center justify-center rounded-full bg-slate-700 sm:size-12"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-user size-4 text-white sm:size-5"
          data-icon="user"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512l388.6 0c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304l-91.4 0z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="flex flex-col space-y-1.5 p-0"
      >
        <h2
          class="leading-none font-semibold group-[&:is(a)]:text-slate-700 group-[&:is(a)]:underline group-[&:is(a):focus]:text-blue-700 group-[&:is(a):hover]:text-blue-700"
        >
          Card Title
        </h2>
        <div
          class="text-sm text-black/60"
        >
          Test Card
        </div>
      </div>
    </div>
    <div
      class="p-6 pt-0 space-y-3"
    >
      <p>
        Card content part 1
      </p>
      <p>
        Card content part 2
      </p>
    </div>
    <div
      class="flex items-center p-6 pt-0"
    >
      <button
        class="font-lato inline-flex items-center justify-center rounded border align-middle outline-offset-4 px-5 py-2.5 text-sm border-gray-300 bg-gray-200 text-slate-700 hover:bg-neutral-300 focus:bg-neutral-300 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70"
      >
        Click me
      </button>
    </div>
  </div>
</div>
`;

exports[`Card > should render a card with a link with an image > expected html 1`] = `
<div>
  <a
    class="group block rounded-xs border bg-white shadow-xs [&:is(a)]:hover:bg-gray-50 [&:is(a)]:hover:shadow-md"
    data-discover="true"
    href="/fr"
  >
    <img
      alt=""
      class="relative aspect-video h-40 w-full object-cover sm:h-48"
      draggable="false"
      loading="lazy"
      src="https://www.canada.ca/content/dam/canada/activities/20250115-1-520x200.jpg"
    />
    <div
      class="flex flex-col space-y-1.5 p-6"
    >
      <h3
        class="leading-none font-semibold group-[&:is(a)]:text-slate-700 group-[&:is(a)]:underline group-[&:is(a):focus]:text-blue-700 group-[&:is(a):hover]:text-blue-700"
      >
        Card Title
      </h3>
      <div
        class="text-sm text-black/60"
      >
        Test Card
      </div>
    </div>
    <div
      class="flex items-center p-6 pt-0"
    >
      <span
        class="inline-block bg-cyan-700 px-2 py-1 text-xs text-white"
      >
        Coming soon
      </span>
    </div>
  </a>
</div>
`;
